### DISCUSSION TITLE
TS error: Using Controller with react-phone-number-input (required value + onChange)

### DISCUSSION THREAD

**Discussion by petr-hajek** [react-hook-form/react-hook-form#2671] [Q&A] [answered]
Hello everyone, thanks for the great library.

I've been trying to plug in PhoneInput component from https://gitlab.com/catamphetamine/react-phone-number-input
It works, but I'm getting TS error: 
`Type '{ as: typeof PhoneInput; name: "phone"; control: Control<{ phone: string; }>; defaultValue: string; className: string; rules: { required: true; validate: (phone: any) => boolean; }; }' is missing the following properties from type 'Pick<PhoneInputProps, "error" | "accept" | "alt" | "autoComplete" | "autoFocus" | "capture" | "checked" | "crossOrigin" | "disabled" | "form" | "formAction" | ... 294 more ... | "useNationalFormatForDefaultCountryValue">': value, onChange`

I'm using this method
```
import PhoneInput from 'react-phone-number-input';
import { Controller } from 'react-hook-form';

<Controller as={PhoneInput} name={"phone"} control={control} defaultValue="" />
```

Compared to the other components used in examples, like the material-ui components, this one has required value & onChange props. Is there a way to update type definitions to allow this?
Edit: using react-hook v6.4.1

**Comment by bluebill1049** [comment, marked as answer]
use `render` prop

**Reply by petr-hajek** [reply]
fair enough, that works, would be great if types allowed to use `as` prop as well, since otherwise it seems to work. Thanks!
